framework/encode/capture_manager.h

```diff
@@ -374,6 +388,178 @@ class CaptureManager
     } rv_annotation_info_;
 };
 
+class ApiCaptureManagerBase
```

Comment:
Another possible idea for breaking apart these classes: Have the common capture manager and each API capture manager be singletons and require the calling code (e.g., in the API-specific capture manager classes, or in the generated encoder files) to know which class to call into. Then we don't need the API capture managers to have a reference to the common manager and the API manager classes don't need to pass-through all the common class methods.

The API capture managers could still store a pointer to the common capture manager for cases when they need to call into it.

Reply:
The goal of the ApiCaptureManagerBase passthrough calls was to minimize the code change impact of call into the common CaptureManager.  Without this, every common call would need to  be changed to be prefixed by common_manager_->  which seemed needlessly noisy (diff-wise) and verbose.

Reply:
Another thought on this: forcing the users of capture managers to be explicit about calling into the API capture manager or the common capture manager may help code clarity. For example, the calls to `AcquireSharedApiCallLock` are currently made on the API-specific capture manager classes `D3D12CaptureManager` or `VulkanCaptureManager`. This could make it not obvious that `D3D12CaptureManager::AcquireSharedApiCallLock` acquires a lock across all capture managers. Making the choice of capture manager explicit could help solidify the concept of global vs API-specific capture functions and promote correct usage.

Reply:
Seems like this makes things more complicated.  Especially because we're still sharing the file write locks.  So now we'd be blocking on API calls and file write locks while the file write itself would be the mitigating restriction.  So just having the file lock would be sufficient.